plugins/pay.c

```diff
@@ -1499,6 +1572,34 @@ static struct command_result *json_paystatus(struct command *cmd,
 		json_array_end(ret);
 		json_object_end(ret);
 	}
+
+	list_for_each(&payments, p, list) {
+		assert(p->parent == NULL);
+		if (b11str && !streq(b11str, p->bolt11))
+			continue;
+
+		json_object_start(ret, NULL);
+		if (p->bolt11)
+			json_add_string(ret, "bolt11", p->bolt11);
+		json_add_u64(ret, "msatoshi",
+			     p->amount.millisatoshis); /* Raw: JSON */
```

Comment:
Yes, you need to use "json_add_msat_only()" and call it "amount_msat".

The rule for JSON output is:
1. Currency amounts must be exposed in millisatoshis.
2. Currency values must be as strings ending in "msat".
3. Currency fieldnames must end in "_msat".

If we'd realized that "millisatoshis everywhere" was a good idea from early on, we could have just done rule 1.
But since we didn't, we need rule 2.  And to help JSON conversion, we need rule 3.